core/perf_test/test_sharedSpace.cpp

```diff
@@ -0,0 +1,300 @@
+
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#if defined _WIN32  // windows system
+#include <windows.h>
+unsigned getBytesPerPage() {
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+}
+#else  // unix/posix system
+#include <unistd.h>
+unsigned getBytesPerPage() { return sysconf(_SC_PAGESIZE); }
+#endif
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <numeric>
+#include <iostream>
+
+namespace {
+void printTimings(std::ostream& out, std::vector<double> const& tr,
+                  size_t numBytes, double threshold) {
+  out << "TimingResult contains " << tr.size() << " results:\n";
+  for (auto it = tr.begin(); it != tr.end(); ++it) {
+    out << "Duration of loop " << it - tr.begin() << " is " << *it
+        << " seconds.";
+    if ((*it) > threshold) {
+      out << " Marked as page migation. ";
+    }
+    out << "The transfer rate is " << numBytes / 1000000000 / (*it)
```

Comment:
```suggestion
    out << " The transfer rate is " << numBytes / 1000000000 / (*it)
```

Reply:
done in [802214](https://github.com/kokkos/kokkos/pull/5289/commits/802214e37b26b5079401cb94b8d5f21a6959152d)